src/game/client/ingameui.cpp

```diff
@@ -1751,3 +1765,1231 @@ void InGameUI::Recreate_Control_Bar()
+
+            for (int i = 0; i < info->tiles_used; i++) {
+                if (m_placeIcon[i] == nullptr) {
+                    m_placeIcon[i] = g_theThingFactory->New_Drawable(m_pendingPlaceType, DRAWABLE_STATUS_8);
+                }
+            }
+
+            for (int i = info->tiles_used; i < g_theWriteableGlobalData->m_maxLineBuildObjects; i++) {
+                if (m_placeIcon[i] != nullptr) {
+                    g_theGameClient->Destroy_Drawable(m_placeIcon[i]);
+                    m_placeIcon[i] = nullptr;
+                }
+            }
+
+            for (int i = 0; i < info->tiles_used; i++) {
+                m_placeIcon[i]->Set_Position(&info->positions[i]);
+                m_placeIcon[i]->Set_Opacity(0.45f);
+                m_placeIcon[i]->Set_Orientation(orientation);
+            }
+        }
+    }
+}
+
+void InGameUI::Draw_Floating_Text()
+{
+    for (auto it = m_floatingTextList.begin(); it != m_floatingTextList.end(); it++) {
+        FloatingTextData *data = *it;
+        int index = g_thePlayerList->Get_Local_Player()->Get_Player_Index();
+        int x;
+        int y;
+        g_thePartitionManager->World_To_Cell(data->m_pos3D.x, data->m_pos3D.y, &x, &y);
+        ICoord2D screen;
+
+        if (g_theTacticalView->World_To_Screen_Tri(&data->m_pos3D, &screen) && data->m_dString != nullptr
+            && g_thePartitionManager->Get_Shroud_Status_For_Player(index, x, y) == SHROUD_STATUS_CLEAR) {
+            screen.y -= data->m_frameCount * m_floatingTextMoveUpSpeed;
+
+            unsigned char red;
+            unsigned char green;
+            unsigned char blue;
+            unsigned char alpha;
+            Get_Color_Components(data->m_color, &red, &green, &blue, &alpha);
+            int color = Make_Color(0, 0, 0, alpha);
+
+            int width;
+            data->m_dString->Get_Size(&width, nullptr);
+            data->m_dString->Draw(screen.x - width / 2, screen.y, data->m_color, color);
+        }
+    }
+}
+
+void InGameUI::Update_And_Draw_World_Animations()
+{
+    for (auto it = m_worldAnimations.begin(); it != m_worldAnimations.end(); it++) {
+        WorldAnimationData *data = *it;
+
+        if (data == nullptr || g_theGameLogic->Is_Game_Paused()) {
+            goto label1;
+        }
+
+        if (g_theGameLogic->Get_Frame() >= (unsigned int)data->m_frame
+            || ((data->m_options & 2) == 0 && (data->m_anim->Get_Status() & Anim2D::STATUS_ANIM_COMPLETE) != 0)) {
+            data->m_anim->Delete_Instance();
+            delete data;
+            it = m_worldAnimations.erase(it);
+        } else {
+            if (data->m_zRise != 0.0f) {
+                data->m_pos.z = data->m_zRise / 30.0f + data->m_pos.z;
+            }
+        label1:
+            int index = g_thePlayerList->Get_Local_Player()->Get_Player_Index();
+
+            if (g_thePartitionManager->Get_Shroud_Status_For_Player(index, &data->m_pos) == SHROUD_STATUS_CLEAR) {

```

Comment:
This does not look right. Above, there is 

```c++
if (data == nullptr || ...)
  goto label1;
```

And here it dereferences `data` when it would be null.

Reply:
Original code is exactly like this and it seems to work.